juju/application.py

```diff
@@ -24,6 +25,42 @@
 
 
 class Application(model.ModelEntity):
+    @property
+    def name(self) -> str:
```

Comment:
Where did these come from previously? Was it falling back to a `__getattr__` in `ModelEntity`?

Reply:
yes that's correct.
and that in turn was .data that is filled in by all watcher deltas